src/main/java/duck/Duck.java

```diff
@@ -0,0 +1,139 @@
+     */
+    public Duck() {
+        this(FILE_PATH);
+    }
+
+
+    /**
+     * Initializes the Duck application by setting up the user interface,
+     * loading tasks from the storage, and displaying startup messages.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Duck(String filePath) {
+        ui = new Ui();
+        tasks = new TaskList();
+        startUpMessage = startUpAndLoadStorage(filePath);
+    }
+
+    /**
+     * Runs the Duck application, which continuously reads and processes user commands
+     * until the exit command is given.
+     */
+    public void run() {
+        ui.sayHi();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String message = ui.readCommand();
+                Command command = Parser.parse(message);
+                command.execute(tasks, storage, ui);
+                isExit = command.isExit();
+            } catch (DuckException e) {
+                ui.displayDukeExceptionMessage(e);
+            }
+        }
+    }
+
+    private String startUpAndLoadStorage(String filePath) {
+        return captureOutput(() -> {
+            try {
+                ui.showStartUpMessage();
+                storage = new Storage(filePath);
+                storage.loadTasks(tasks);
+                ui.showStartUpCompleteMessage();
+            } catch (DuckException e) {
+                ui.displayDukeExceptionMessage(e);
+            }
+        });
+    }
+
+    /**
+     * Returns the startup message generated by the Duck application.
+     *
+     * @return The startup message generated by the Duck application.
+     */
+    public String getStartUpMessage() {
+        return startUpMessage;
+    }
+
+    /**
+     * Returns the response generated by the Duck application for the given user input.
+     *
+     * @return The response generated by the Duck application.
+     */
+    public String getResponse(String message) {
+        // Create a ByteArrayOutputStream to capture the output
+        return captureOutput(() -> {
+            try {
+                Command command = Parser.parse(message);
+                command.execute(tasks, storage, ui);
+            } catch (DuckException e) {
+                ui.displayDukeExceptionMessage(e);
+            }
+        });
+    }
+
+    private String captureOutput(Runnable runnable) {
+        // Create a ByteArrayOutputStream to capture the output
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+
+        PrintStream originalOut = System.out; // Save the original System.out
+        System.setOut(new PrintStream(baos)); // Redirect System.out to the ByteArrayOutputStream
+
+        runnable.run();
+
+        // Ensure all output is flushed
+        System.out.flush();
+
+        // Convert the captured output to a string
+        return baos.toString();
+    }
```

Comment:
This is a good and interesting approach to capturing the output. Nicely done!

Reply:
> This is a good and interesting approach to capturing the output. Nicely done!
Thanks! I thought this used extraction method to improve code readability, but unfortunately it still violates SLAP :(.